service-worker/worker/src/plugins/external/index.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  FetchInstruction,
+  Operation,
+  Plugin,
+  PluginFactory,
+  VersionWorker,
+  VersionWorkerImpl,
+  cacheFromNetworkOp,
+  fetchFromCacheInstruction,
+} from '@angular/service-worker/worker';
+
+interface UrlConfig {
+  url: string;
+}
+
+interface ExternalManifest {
+  urls: UrlConfig[];
+}
+
+export interface ExternalContentCacheOptions {
+  manifestKey?: string;
+}
+
+export function ExternalContentCache(options?: ExternalContentCacheOptions): PluginFactory<ExternalPlugin> {
+  const manifestKey = (options && options.manifestKey) || 'external';
+  return (worker: VersionWorker) => new ExternalPlugin(worker as VersionWorkerImpl, manifestKey);
+}
+
+export class ExternalPlugin implements Plugin<ExternalPlugin> {
+  private cacheKey: string;
+
+  constructor(public worker: VersionWorkerImpl, public key: string) {
```

Comment:
Originally there was a planned distinction between the public interface of the `VersionWorker` presented to plugins and the `VersionWorkerImpl` implementation actually used. There turned out to be no practical benefit to this, though, so the `VersionWorker` interface is really legacy and needs to be dropped entirely. Everything in practice is done with the concrete type.